docs/source/contributing.md

```diff
@@ -2,154 +2,88 @@
-
-```bash
-git fetch --all
-git rebase upstream/master
-```
-
-Rebasing is a very powerful feature of Git. You need to understand how it works or else you will risk losing your work. Read about it in the [Git documentation](https://git-scm.com/docs/git-rebase), it will be well worth it. In a nutshell, rebasing does the following:
-- "Unwinds" your local commits. Your local commits are removed temporarily from the history.
-- The latest changes from upstream are added to the history
-- Your local commits are re-applied one by one
-- If there are merge conflicts, you will be prompted to fix them before continuing. Read the output closely. It will tell you how to complete the rebase.
-- When done rebasing, you will see all of your commits in the history.
+In case of bigger features or API changes, it's required to prepare an enhancement proposal beforehand as usually these require broader discussions.
+For further details, check https://github.com/scylladb/scylla-operator/tree/master/enhancements.
 
 ## Submitting a Pull Request
 
-Once you have implemented the feature or bug fix in your branch, you will open a PR to the upstream repo. Before opening the PR ensure you have added unit tests, are passing the integration tests, cleaned your commit history, and have rebased on the latest upstream.
-
-In order to open a pull request (PR) it is required to be up to date with the latest changes upstream. If other commits are pushed upstream before your PR is merged, you will also need to rebase again before it will be merged.
+Once you have implemented the feature or bug fix in your branch, you need to open a PR to the upstream repo.
+Before opening the PR ensure you have added unit tests, all e2e's are passing, and your commit history is clean.
 
 ### Commit History
 
-To prepare your branch to open a PR, you will need to have the minimal number of logical commits so we can maintain
-a clean commit history. Most commonly a PR will include a single commit where all changes are squashed, although
-sometimes there will be multiple logical commits.
-
-```bash
-# Inspect your commit history to determine if you need to squash commits
-git log
-
-# Rebase the commits and edit, squash, or even reorder them as you determine will keep the history clean.
-# In this example, the last 5 commits will be opened in the git rebase tool.
-git rebase -i HEAD~5
-```
-
-Once your commit history is clean, ensure you have based on the [latest upstream](#updating-your-fork) before you open the PR.
+To prepare your branch to open a PR, you will need to have a minimal number of logical commits so we can maintain
+a clean commit history. Most commonly, a PR includes a single commit where all changes are squashed and a separate commit for autogenerated changes.
 
-### Commit messages
+### Commits and PRs
```

Comment:
It would be good to mentions the imperative mood of commits aka `if applied this commit will ...`
e..g https://cbea.ms/git-commit/#imperative